plugins/catalog-backend/src/search/DefaultCatalogCollator.ts

```diff
@@ -77,3 +95,18 @@ export class DefaultCatalogCollator implements DocumentCollator {
     });
   }
 }
+
+export function mapFilterToQueryString(
```

Comment:
The filter serialization is actually a bit more involved than this.

https://github.com/backstage/backstage/blob/1fa98590a9629e8e9dc399b21f44a20139a38497/packages/catalog-client/src/CatalogClient.ts?plain=1#L69

I think to make this work, you really should look into replacing the raw `fetch` with using `CatalogClient.getEntities({ this.filter })` from `@backstage/catalog-client`. That's what everybody does that wants to talk to the catalog :)

Reply:
Oh btw, please don't apologise for having a lot of notes. I'm new to the codebase so I'm still feeling my way around, and figuring out how to do things in an idiomatic way. If anything, it illustrates the quality level you want to have in the codebase :)